src/sth_database.js

```diff
@@ -146,19 +151,15 @@
    * @param {string} resolution The resolution of the data to use
    * @param {Date} from The date from which retrieve the aggregated data
    * @param {Date} to The date to which retrieve the aggregated data
+   * @param {boolean} shouldFilter If true, the null results are filter out
    * @param {Function} callback Callback to inform about any possible error or results
    */
-  function getAggregatedData(collectionName, aggregatedFunction, resolution, from, to, callback) {
+  function getAggregatedData(collectionName, aggregatedFunction, resolution, from, to, shouldFilter, callback) {
     var fieldFilter = {
-      '_id': 0,
-      'resolution': 1,
-      'range': 1,
-      'origin': 1,
-      'type': 1,
-      'values.offset': 1,
-      'values.samples': 1
+      'points.offset': 1,
+      'points.samples': 1
     };
-    fieldFilter['values.' + aggregatedFunction] = 1;
+    fieldFilter['points.' + aggregatedFunction] = 1;
```

Comment:
Yeah, sorry! Just to avoid using the term "values" already imposed by NGSI. This is an example of the responses we are currently providing. Take a look at the use of the "values" and "points" keys:
{"contextResponses":[{"contextElement":{"attributes":[{"name":"attributeId","values":[{"_id":{"origin":"2015-02-22T16:19:00.000Z","range":"minute","resolution":"second"},"points":[{"offset":9,"samples":1,"sum":34.96}]}]}],"id":"entityId","isPattern":false},"statusCode":{"code":"200","reasonPhrase":"OK"}}]}